go-selinux/selinux_linux.go

```diff
@@ -237,20 +235,21 @@ func readConfig(target string) string {
 	scanner := bufio.NewScanner(in)
 
 	for scanner.Scan() {
-		line := strings.TrimSpace(scanner.Text())
-		if len(line) == 0 {
+		str := bytes.TrimSpace(scanner.Bytes())
+		if len(str) == 0 {
 			// Skip blank lines
 			continue
 		}
-		if line[0] == ';' || line[0] == '#' {
+		if str[0] == ';' || str[0] == '#' {
 			// Skip comments
 			continue
 		}
-		if groups := assignRegex.FindStringSubmatch(line); groups != nil {
-			key, val := strings.TrimSpace(groups[1]), strings.TrimSpace(groups[2])
-			if key == target {
-				return strings.Trim(val, "\"")
-			}
+		fields := bytes.SplitN(str, []byte{'='}, 2)
```

Comment:
Just noticing this: this also accepts `=` inside the value part;  this was already the case in the regex, but mostly wondering what values are accepted (white-space included if there's no quotes? `=` included?)

Not for this PR, but perhaps something that needs to be checked (in a follow-up)

Reply:
We're using `SplitN` so only the first (i.e. leftmost) `=` is used for splitting. Now, variable name can not contain `=` so we're good here.

Whitespace is not accepted (around `=`) because the same file is also parsed by shell, which does not allow spaces around `=`.

Reply:
Yeah, we can probably continue to just assume "whatever comes after `=` (minus quotes) _should_ be valid". It was mostly an observation.